analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -887,6 +868,30 @@ def main(args):
     # logged by the logger.
     all_cmp_cmd_count = len(compile_commands)
 
+    # We clear the output directory in the following cases.
+    ctu_dir = os.path.join(args.output_path, 'ctu-dir')
+    if 'ctu_phases' in args and args.ctu_phases[0] and \
+            os.path.isdir(ctu_dir):
+        # Clear the CTU-dir if the user turned on the collection phase.
+        LOG.debug("Previous CTU contents have been deleted.")
+        shutil.rmtree(ctu_dir)
+
+    if 'clean' in args and os.path.isdir(args.output_path):
+        LOG.info("Previous analysis results in '%s' have been removed, "
+                 "overwriting with current result", args.output_path)
+        shutil.rmtree(args.output_path)
+
+    if not os.path.exists(args.output_path):
+        os.makedirs(args.output_path)
+
+    # TODO: I'm not sure that this directory should be created here.
```

Comment:
I think we can move these instruction down above the `perform_analysis` function call. For example if no analysis is required below we will exit immediately but we will create an empty fixit directory.

Reply:
This was the lowermost position, because the next function call is `parse_unique_logs()` uses these directories.